engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -465,6 +471,49 @@ private Class<?> findNestedClass(Class<?> enclosingClass, String nestedClassName
         return m.get(nestedClassName);
     }
 
+    private Method getMethod(final Class<?> scope, final MethodCallExpr methodCallExpr, final Class<?>[] paramTypes,
+            final Class<?>[][] parameterizedTypes) {
+        final String methodName = methodCallExpr.getNameAsString();
+        final QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+
+        if (scope == null) {
+            final Class<?> methodClass = variables.get(methodName);
+            if (methodClass == PyCallableWrapper.class) {
+                final Object paramValueRaw = queryScope.readParamValue(methodName, null);
+                final PyCallableWrapper pyCallableWrapper = (PyCallableWrapper) paramValueRaw;
+                try {
+                    String javaMethodName = pyCallableWrapper.getAttribute("_j_simple_name", String.class);
+                    methodCallExpr.setName(javaMethodName);
+                    return getMethod(null, javaMethodName, paramTypes, parameterizedTypes);
+                } catch (RuntimeException e) {
+                    // Not, swappable function, can safely ignore
+                }
+            }
+        } else if (scope == org.jpy.PyObject.class || scope == PyCallableWrapper.class) {
+            String pythonExpr = methodCallExpr.getScope().get().toString() + "." + methodName;
+            PythonDeephavenSession pds = (PythonDeephavenSession) ((AbstractScriptSession.UnsynchronizedScriptSessionQueryScope) queryScope).scriptSession();
+            Map<String, PyObject> scopeVars = pds.getVariablesRaw();
+            try {
+                // To cover all possible use cases, we need to 'unwrap' all the other wrapped vars (PyCallable Wrapper,
+                // PyListWrapper,
+                // PyDictWrapper) in the scope before passing them to executeCode. But the chance of a user setting
+                // a swappable function on the list/dict-like object is very low.
+                PyObject pyobj = PyObject.executeCode(pythonExpr, PyInputMode.EXPRESSION, scopeVars, null);
+                Object obj = PythonScopeJpyImpl.convert(pyobj);
+                if (obj.getClass() == (PyCallableWrapper.class)) {
+                    String javaMethodName = ((PyCallableWrapper) obj).getAttribute("_j_simple_name", String.class);
+                    methodCallExpr.setScope(null);
+                    methodCallExpr.setName(javaMethodName);
+                    return getMethod(null, javaMethodName, paramTypes, parameterizedTypes);
+                }
+            } catch (Exception e) {
+                // to report a compilation error?
```

Comment:
perhaps a warn or some other logging, so we know we couldn't magically make it java?